### DISCUSSION TITLE
Multi-server support in UI

### DISCUSSION THREAD

**Discussion by ivdimitro** [zwave-js/zwave-js-ui#3264] [Ideas]
I know it will be a niche use case but have you considered to add support for for multiple zwaveJS controllers/servers in the UI? At least I am kind of forced to use two controllers due to a thick concrete wall across my apartment and not enough routers. I am using mainly sensors on zwave. 

The way I see it, it should relatively easy - for example add one more column in the UI to indicate under which controller are the particular devices and the rest can be the same. 
In the map we will need a bit more work to represent the two separate controllers but if it is difficult you can represent then in separate views. 
I guess there are other

**Comment by robertsLando** [comment]
This would not be that easy unfortunately and giving that zui can be used with docker it's far easier to just have multiple instances of it running on differents ports

**Comment by polskikrol** [comment]
Well, is this still the case? I have a spare USB Aeotec controller and want to upgrade the firmware on it. Is it possible to support more than one USB controller, whereas one is part of the network and nodes, and the other just 'shows up' unprovisioned in the GUI? (Long term, would be kinda neat whereas you can provision or promote the spare to Primary Controller by automating the NVM backup, search and replace, and load operation. Then just dump the old controller into the 'unprovisioned' area again.
